pkg/cached/cache.go

```diff
@@ -55,51 +57,65 @@ import (
 )
 
 // Result is the content returned from a call to a cache. It can either
-// be created with [NewResultOK] if the call was a success, or
-// [NewResultErr] if the call resulted in an error.
+// be created with [NewResult] if the call was a success, or
+// [NewError] if the call resulted in an error.
 type Result[T any] struct {
 	Data T
 	Etag string
 	Err  error
 }
 
-// NewResultOK creates a new [Result] for a successful operation.
-func NewResultOK[T any](data T, etag string) Result[T] {
+// NewResult creates a new [Result] for a successful operation.
+func NewResult[T any](data T, etag string) Result[T] {
 	return Result[T]{
 		Data: data,
 		Etag: etag,
 	}
 }
 
-// NewResultErr creates a new [Result] when an error has happened.
-func NewResultErr[T any](err error) Result[T] {
+// NewError creates a new [Result] when an error has happened.
+func NewError[T any](err error) Result[T] {
 	return Result[T]{
 		Err: err,
 	}
 }
 
-// Result can be treated as a [Data] if necessary.
+// Errorf creates a new [Result] when an error has happened.
+func Errorf[T any](s string, params ...any) Result[T] {
+	return NewError[T](fmt.Errorf(s, params...))
+}
+
+// Get returns the cached result. Result can be treated as a [Data] if necessary.
 func (r Result[T]) Get() Result[T] {
 	return r
 }
 
+// Unpack return the individual elements of the result, for more
+// convenient usage:
+//
+//	if value, _, err := c.Get().Unpack(); err != nil {
+//	  ...
+//	}
+func (r Result[T]) Unpack() (T, string, error) {
```

Comment:
though it's pretty intuitive, would be nice to see an example of usage in the tests

Reply:
Tuples are idiomatic for return values, not structs that include errors. This lets us stay with tubles, but have a nice way to call it in a Go like way:
```golang
if value, etag, err := something.Get().Unpack(); err != nil {
  ...
}
```
Adding it to the tests.

Reply:
Alternative: `Get() (value T, etag string, err error)`. Which one do you prefer?

Reply:
have done the variant with `Get` now. IMO this is much more natural.